ui/raidboss/data/05-shb/alliance/the_tower_at_paradigms_breach.ts

```diff
@@ -468,6 +477,144 @@ const triggerSet: TriggerSet<Data> = {
+      delaySeconds: 0.25,
+      durationSeconds: 5,
+      suppressSeconds: 0.5,
+      alertText: (data, _matches, output) => {
+        if (!data.deployArmaments)
+          return;
+
+        // define some expected results
+        const verticalMiddle = (1 << 1 | 1 << 4 | 1 << 7);
+        const verticalSides = (1 << 0 | 1 << 2 |
+          1 << 3 | 1 << 5 |
+          1 << 6 | 1 << 8);
+        const horizontalMiddle = (1 << 3 | 1 << 4 | 1 << 5);
+        const horizontalSides = (1 << 0 | 1 << 1 | 1 << 2 |
+          1 << 6 | 1 << 7 | 1 << 8);
+        const nsMiddle = (1 << 1 | 1 << 7);
+        const ewMiddle = (1 << 3 | 1 << 5);
+        const corners = (1 << 0 | 1 << 2 | 1 << 6 | 1 << 8);
+        const center = (1 << 4);
+
+        // Start with all spots marked as safe...
+        let safeSpots = (1 << 0 | 1 << 1 | 1 << 2 |
+          1 << 3 | 1 << 4 | 1 << 5 |
+          1 << 6 | 1 << 7 | 1 << 8);
+
+        // Then figure out which spots are unsafe due to pending AOEs...
+
+        if (data.deployArmaments.includes(DeployArmaments.VerticalMiddle)) {
+          // Clear the vertical middle column
+          safeSpots &= ~verticalMiddle;
+        }
+
+        if (data.deployArmaments.includes(DeployArmaments.VerticalSides)) {
+          // Clear the vertical side columns
+          safeSpots &= ~verticalSides;
+        }
+
+        if (data.deployArmaments.includes(DeployArmaments.HorizontalMiddle)) {
+          // Clear the horizontal middle row
+          safeSpots &= ~horizontalMiddle;
+        }
+
+        if (data.deployArmaments.includes(DeployArmaments.HorizontalSides)) {
+          // Clear the horizontal side rows
+          safeSpots &= ~horizontalSides;
+        }
+
+        // Finally, determine where to alert
+        if (
+          safeSpots === verticalMiddle ||
+          safeSpots === horizontalMiddle ||
+          safeSpots === center
+        )
+          return output.center!();
+        else if (safeSpots === verticalSides)
+          return output.ewSides!();
+        else if (safeSpots === horizontalSides)
+          return output.nsSides!();
+        else if (safeSpots === nsMiddle)
+          return output.nsMiddle!();
+        else if (safeSpots === ewMiddle)
+          return output.ewMiddle!();
+        else if (safeSpots === corners)
+          return output.corners!();
+
+        return output.oops!();
+      },
+      run: (data) => delete data.deployArmaments,
+      outputStrings: {
+        center: {
+          en: 'Go Center',
+        },
+        ewSides: {
+          en: 'Go E/W Sides',
+        },
+        nsSides: {
+          en: 'Go N/S Sides',
+        },
+        ewMiddle: {
+          en: 'Go E/W Middle',
+        },
+        nsMiddle: {
+          en: 'Go N/S Middle',
+        },
+        corners: {
+          en: 'Go Corners',
+        },
+        oops: {
+          en: 'Avoid line AOEs',
+        },
```

Comment:
For clarity, this is just here as a fallback in the weird case that we get something which we don't understand

Reply:
Makes sense to me!